### PR TITLE
Use fifo close on task exec cleanup to prevent directory leak

### PR DESCRIPTION
**Pull request by austinvazquez**
*Issue #, if available:*
N/A

*Description of changes:*
Close fifoSet for the task and exec when the task is deleted.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by djdongjin — APPROVED** [review]

**Comment by austinvazquez** [comment]
@djdongjin updated commit message and PR title to be more specific to the leak. Please re-review.

**Review by djdongjin — APPROVED** [review]

File: agent/service.go [hunk 1]
```diff
@@ -253,6 +253,9 @@ func (ts *TaskService) Create(requestCtx context.Context, req *taskAPI.CreateTas
 			logger.WithError(err).Error()
 			return nil, err
 		}
+		ts.addCleanup(taskExecID, func() error {
+			return fifoSet.Close()
```
Comment (swagatbora90) [inline agent/service.go:257]: These are fifos inside the VM, right? Once the task gets deleted, the VM should be shut down, so how is this contributing to the leak?

Comment (austinvazquez) [reply agent/service.go:257]: > Once the task gets deleted, the VM should be shut down

The VM _can_ shutdown when all tasks are stopped, but that is not always the case. Default is for VM to stay up even if all tasks are deleted. See https://github.com/firecracker-microvm/firecracker-containerd/blob/main/proto/firecracker.proto#L34

However, you are correct `Create` cleanup removes the top level directory here and wouldn't contribute to the leak. Making this not _required_ but doesn't hurt anything to have either. Would you prefer to exclude this?

Comment (swagatbora90) [reply agent/service.go:257]: Okay, just wanted to understand if this was contributing to the leak as well besides the exec path. I am fine keeping this in as an extra safety measure.

**Review by swagatbora90 — APPROVED** [review]
LGTM

**Pull request merged**